File: abc/abc175/abc175_c.fs
// https://atcoder.jp/contests/abc175/submissions/15967342

let [| X; K; D |] = stdin.ReadLine().Split() |> Array.map int64
let absX = abs X
let r = min K (absX / D)
let X1 = absX - r * D

if (K - r) % 2L = 1L then abs (X1 - D)
else X1
|> stdout.WriteLine

File: abc/abc175/abc175_c.cs
// https://atcoder.jp/contests/abc175/submissions/15967505

using System;
using System.Linq;

class Program
{
    static void Main()
    {
        var input = Console.ReadLine().Split().Select(long.Parse).ToArray();
        var X = Math.Abs(input[0]);
        var K = input[1];
        var D = input[2];
        var r = Math.Min(K, X / D);
        var X1 = X - r * D;

        if ((K - r) % 2 == 1)
        {
            Console.WriteLine(Math.Abs(X1 - D));
        }
        else
        {
            Console.WriteLine(X1);
        }
    }
}
